src/schema/v2/artist/createArtistMutation.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  GraphQLBoolean,
+  GraphQLNonNull,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { GraphQLObjectType } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+import { snakeCase } from "lodash"
+import { ResolverContext } from "types/graphql"
+import { ArtistType } from "./index"
+
+interface CreateArtistMutationInput {
+  birthday: string
+  deathday: string
+  displayName: string
+  first: string
+  isPersonalArtist: boolean
+  last: string
+  middle: string
+  nationality: string
+}
+
+const CreateArtistSuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "CreateArtistSuccess",
+  isTypeOf: (data) => {
+    return data._type !== "GravityMutationError"
+  },
+  fields: () => ({
+    artist: {
+      type: ArtistType,
+      resolve: (artist) => artist,
+    },
+  }),
+})
+
+const CreateArtistFailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "CreateArtistFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const SuccessOrErrorType = new GraphQLUnionType({
+  name: "CreateArtistSuccessOrErrorType",
+  types: [CreateArtistSuccessType, CreateArtistFailureType],
+})
+
+export const createArtistMutation = mutationWithClientMutationId<
+  CreateArtistMutationInput,
+  any | null,
+  ResolverContext
+>({
+  name: "CreateArtistMutation",
+  description: "Create an artist",
+  inputFields: {
+    birthday: { type: GraphQLString },
+    deathday: { type: GraphQLString },
+    displayName: { type: new GraphQLNonNull(GraphQLString) },
+    first: { type: GraphQLString },
+    isPersonalArtist: { type: GraphQLBoolean },
```

Comment:
**thought:** What would happen if we set this to `false`? I believe we should not allow this argument and instead set it by default in Gravity.

Reply:
If we do not pass this field, then users cannot create Artist. Gravity expects this param. Only Partners can skip passing this param.

Reply:
I would then inject it from Metaphysics to Gravity without passing it from the client side. I would try to limit the args sent from the client to only the ones we really need

Reply:
Injecting it would automatically mean that a create artist mutation only creates only personal artists for users only. In that case maybe it should not be named createArtistMutation, instead it should be named createPersonalArtistMutation. I will private message you as I cannot explain more here without revealing too much backend details.

Reply:
sounds good - I think you get what I mean though. I was more referring to a `createPersonalArtistMutation` behaviour rather than a `createArtistMutation`